pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -140,6 +141,9 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 
 	externalService, err := common.ReconcileService(ctx, d.Client, services.NewExternalService(d.ES), &d.ES)
 	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return results.WithReconciliationState(defaultRequeue.WithReason("Could not reconcile external service"))
```

Comment:
Maybe something along the lines of
```suggestion
			return results.WithReconciliationState(defaultRequeue.WithReason("Pending external Service recreation"))
```

Reply:
Or maybe be explicit about the name of the service, because I think the concept of "external" vs "internal" service is not something users are aware of. We don't mention it in the docs afaik. It is a code level concept only. 

```diff
diff --git a/pkg/controller/elasticsearch/driver/driver.go b/pkg/controller/elasticsearch/driver/driver.go
index df3defd16..434636b04 100644
--- a/pkg/controller/elasticsearch/driver/driver.go
+++ b/pkg/controller/elasticsearch/driver/driver.go
@@ -142,7 +142,7 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	externalService, err := common.ReconcileService(ctx, d.Client, services.NewExternalService(d.ES), &d.ES)
 	if err != nil {
 		if apierrors.IsAlreadyExists(err) {
-			return results.WithReconciliationState(defaultRequeue.WithReason("Could not reconcile external service"))
+			return results.WithReconciliationState(defaultRequeue.WithReason(fmt.Sprintf("Pending %s service recreation", services.ExternalServiceName(d.ES.Name))))
 		}
 		return results.WithError(err)
 	}
```